pkg/util/cidr/cidr.go

```diff
@@ -0,0 +1,297 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cidr
+
+import (
+	"context"
+	"encoding/json"
+	io "io"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/settings"
+	"github.com/cockroachdb/cockroach/pkg/util/envutil"
+	cjson "github.com/cockroachdb/cockroach/pkg/util/json"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/stop"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+)
+
+var cidrMappingUrl = settings.RegisterStringSetting(
+	settings.ApplicationLevel,
+	"metrics.cidr_mapping_url",
+	"url of a JSON file containing a list of CIDR blocks (file:// or http://)",
+	envutil.EnvOrDefaultString("COCKROACH_CIDR_MAPPING", ""),
+	settings.WithPublic,
+)
```

Comment:
Will it be possible to declare multiple `cidr_mapping_url`s?

* If we go the route of pointing CRDB directly to each cloud's IP range list, then we'll have three different URLs to declare. Here, if we _don't_ support multiple files, then CC will never need to use `cidr_mapping_transform`, and would go with the local file route.
* If we go the route of pointing CRDB to a local file, it may happen that that file is too large to (easily) propagate it to the nodes file system. Granted, we can probably work around the latter restriction ([thread](https://cockroachlabs.slack.com/archives/C07597E1JGZ/p1720638878877559)), but wanted to check

Reply:
It would be possible to have multiple URLs, but I wasn't sure if this is the right approach. I also agree we probably won't end up using the transform, and it would be easy to remove it. It made some of the testing easier, but other than that, if we go the file route it doesn't make much difference. 

There are various ways we can do this in the end, initially I was trying to add enough flexibility to allow you to try out some different ways and we can clean it up before we merge the PR.

In terms of multiple files, that would also be possible. An alternative is if the pointed to file is a directory (or regex), the code could scan the directory (or regex) and load all files that match. 

All of these options are pretty easy and self-contained in that one commit, so I'm not too concerned about changing it.

Reply:
I'm in favor of pointing to a single file in GCS. It seems less complicated than multiple files, compression, etc. I'll work towards that and see how things shake out